### ISSUE TITLE
support for sub-directories

### ISSUE THREAD

**Issue by Nkmol** [Microsoft/vscode-tslint#228]
As noted with #222 , it seems that `vscode-tslint` does not work correctly with sub-directories.

An example map structure:
```
project
   app
   server
```

Where my `app` folder uses the `tslint`, and the server uses different lintings.

[I reported this to the `tslint` github](https://github.com/palantir/tslint/issues/2973), as it was indicated `vscode-tslint` uses the `tslint` API for this.

@ajafff responded on this with:

> I looked into this a bit more. It seems like `vscode-tslint` doesn't even try to load `tslint` when it is installed as dependency in a subdirectory when opening a file in that directory.

**Comment by egamma** [comment]
@Nkmol I assume you only have a tslint.json inside the app folder?

People were complaining that tslint is loaded to eagerly, to address this I no longer load tslint, when it doesn't find a tslint.json at the root of the workspace. 

To verify that this is the case can you move the tslint.json to the root of the workspace?

**Comment by Nkmol** [comment]
@egamma Thank you for your comment!

> I assume you only have a tslint.json inside the app folder?

Yes, my `app` folder is its own NPM project with the `tslint.json` file.

> To verify that this is the case can you move the tslint.json to the root of the workspace?

Placing the `tslint.json` in the root of workspace requires me to create a new NPM project, with a new install of the `tslint` module. 

It seems to work just fine this way, with a simple "activation" config:

**tslint.json**
``` typescript
{

}
```
<hr />

But the reason that I wanted everything in one map, is pure for encapsulation. 
What if only one project uses typescript, then it would be really confusing to put these packages in the root map. 
If multiple projects would use this, it _would_ be helpful to create one 'base' `tsconfig.json` in the root map. 
Now, with one linting it is easy to follow. But imagine having to do this for every linting (ES, TS, SCSS, LESS, etc). Would it not be really confusing to have all of these lintings in the root map, and be loaded when you probably only need one or two of them?

Keeping the flexibility of working on multiple projects at the same time, but also able to focus on one particular project when needed, is the flexibility I want to go for.

I am not sure how `vscode-eslint` does this now, but it seems to work this way now.

<hr />

Maybe this goes outside the scope of `vscode-tslint` only 😕 

I hope the example I gave is understandable and is seen as a valid use-case.

**Comment by Nkmol** [comment]
@egamma Thank you for your comment!

> I assume you only have a tslint.json inside the app folder?

Yes, my `app` folder is its own NPM project with the `tslint.json` file.

> People were complaining that tslint is loaded to eagerly, to address this I no longer load tslint, when it doesn't find a tslint.json at the root of the workspace.

It does seem to work differently. Without using a `tslint.json` in the root of the workspace
  and with simply installing the `tslint` node_module inside the root of the workspace. It seems to work perfectly:

![image](https://user-images.githubusercontent.com/8686001/28022997-e6790362-658c-11e7-89b1-2f398fe4e3c1.png)

<hr />

What if only one project uses typescript, then it would be really confusing to put these packages in the root map. Especially if someone is not using any of these packages, it breaks the encapsulation and gives unnecessary information to the developer.

Keeping the flexibility of working on multiple projects at the same time, but also able to focus on one particular project when needed, is the flexibility I want to go for.

_It would be really cool, if once you open a `.ts` file, the nearest tslint node_module is loaded._

I am not sure how `vscode-eslint` does this now, but it seems to work this way now. (https://github.com/Microsoft/vscode-eslint/issues/32)

<hr />

I hope the example I gave is understandable and is seen as a valid use-case.

Once again, thank you for your time!

**Comment by egamma** [comment]
@Nkmol actually I think you could also use `tslint.nodePath` in your setup as described in this comment https://github.com/Microsoft/vscode-tslint/issues/184#issuecomment-291524064

**Comment by Nkmol** [comment]
@egamma Wow, exactly what I was looking for!

Though, having multiple typescript projects with different `tslint.config` would not work this way. 
Probably I would create an NPM project in the workspace and create a 'base' `tslint.config` to load from.

This solves my issue, thank you 😄
